tests/backends/test_backends.py

```diff
@@ -0,0 +1,160 @@
+from dlup.writers import TiffCompression, TifffileImageWriter
+
+
+@pytest.fixture
+def file_path(tmp_path):
+    # tmp_path is a pytest fixture that provides a Path object to a temporary directory.
+    # Here we are appending a file name to that path.
+    return tmp_path / "test_image.tif"
+
+
+def write_image_to_tiff(file_path, image, mpp, size, pyramid):
+    array = np.asarray(image)
+    channels = array.shape[2] if array.ndim == 3 else 1
+    writer = TifffileImageWriter(
+        file_path,
+        size=(*size[::-1], channels),
+        mpp=mpp,
+        compression=TiffCompression.NONE,
+        interpolator=Resampling.NEAREST,
+        tile_size=(128, 128),
+        pyramid=pyramid,
+    )
+    writer.from_pil(image)
+
+
+def create_test_image(size, channels, color1, color2):
+    half_size = (size[0] // 2, size[1] // 2)
+    if channels == 3:
+        array = np.zeros((*size, 3), dtype=np.uint8)
+        array[: half_size[0], : half_size[1], :] = color1
+        array[half_size[0] :, half_size[1] :, :] = color2
+    else:
+        array = np.zeros(size, dtype=np.uint8)
+        array[: half_size[0], : half_size[1]] = color1
+        array[half_size[0] :, half_size[1] :] = color2
+    return PIL.Image.fromarray(array, mode="RGB" if channels == 3 else "L")
+
+
+@pytest.fixture
+def slides(file_path, test_image, mpp, size, pyramid):
+    # Write the test image to a TIFF file
+    write_image_to_tiff(str(file_path), test_image, mpp, size, pyramid)
+    # Open the written TIFF file with both backends
+    tiff_slide = open_slide_tifffile(str(file_path))
+    openslide_slide = open_slide_openslide(str(file_path))
+    yield tiff_slide, openslide_slide
+    # After the test function that uses this fixture finishes, close both slides
+    tiff_slide.close()
+    openslide_slide.close()
+
+
+@pytest.fixture
+def test_image(size, channels):
+    color1 = 255
+    color2 = 127
+    # Assuming 'create_test_image' is a static method or standalone function
+    return create_test_image(size, channels, color1, color2)
+
+
+@pytest.fixture
+def mpp():
+    return (1.0, 0.999)
+
+
+class TestBackends:
+    def read_region_and_properties_asserts(self, tiff_slide, openslide_slide, mode, size, mpp, pyramid, test_image):
+        original_array = np.asarray(test_image)
+        properties = tiff_slide.properties
+        tile_size = (properties["tifffile.level[0].TileWidth"], properties["tifffile.level[0].TileLength"])
+        num_levels = int(np.ceil(np.log2(np.asarray(size[::-1]) / np.asarray(tile_size))).min()) + 1
+
+        # Let's try to read outside of the slide levels. This should give a RuntimeError
+        with pytest.raises(RuntimeError):
+            tiff_slide.read_region((0, 0), num_levels + 1, (1, 1))
+
+        # We need to check a few regions to make sure the backend is working correctly
+        # 1. Check the whole image
+        # 2. Check the top right corner
+        # 3. Check the bottom left corner
+
```

Comment:
Why is this the top right and bottom left corner? It seems to me that the code corresponds to 2. the top left corner and 3. the middle